addons/web/static/src/views/fields/monetary/monetary_field.js

```diff
@@ -77,9 +77,9 @@ MonetaryField.defaultProps = {
 MonetaryField.supportedTypes = ["monetary", "float"];
 MonetaryField.displayName = _lt("Monetary");
 
-MonetaryField.extractProps = ({ attrs }) => {
+MonetaryField.extractProps = ({ attrs, field }) => {
     return {
-        currencyField: attrs.options.currency_field,
+        currencyField: attrs.options.currency_field || field.currency_field || "currency_id",
```

Comment:
would do the fallbacks in the currencyId getter, as this method is about extracting info from the node (in the arch).

Reply:
I tried to follow the existing logic of the float field, with the `digits` attribute, but if you prefer a change in the getter, both works for me:
https://github.com/odoo/odoo/blob/f1a9de85c19a53c9516cac44de6cbaf53d850842/addons/web/static/src/views/fields/float/float_field.js#L51-L59

Reply:
Yep, the rationale is that we don't necessarily go through `extractProps`. This method is there to read the attrs from an arch and derive props from them. If you want to use the FloatField in a client action for instance, you don't have an arch but you'd like the fallback to be done anyway.